eel/src/data_store.rs

```diff
@@ -73,20 +74,48 @@ impl DataStore {
                 snapshot_id,
             ),
         )
-        .map_to_invalid_input("Failed to add new incoming payment to payments db")?;
+		.map_to_invalid_input("Failed to add new incoming payment to payments db")?;
+        let payment_id = tx.last_insert_rowid();
         tx.execute(
             "\
             INSERT INTO events (payment_id, type, timezone_id, timezone_utc_offset_secs) \
             VALUES (?1, ?2, ?3, ?4) \
             ",
             (
-                tx.last_insert_rowid(),
+                payment_id,
                 PaymentState::Created as u8,
                 &self.timezone_config.timezone_id,
                 self.timezone_config.timezone_utc_offset_secs,
             ),
         )
         .map_to_invalid_input("Failed to add new incoming payment to payments db")?;
+
+        #[allow(clippy::single_match)]
+        match offer_kind {
+            Some(OfferKind::Pocket {
+                id: pocket_id,
+                exchange_rate:
+                    ExchangeRate {
+                        currency_code,
+                        rate,
+                        updated_at,
+                    },
+                topup_value_minor_units,
+                exchange_fee_minor_units,
+                exchange_fee_rate_permyriad,
+            }) => {
+                let exchanged_at: DateTime<Utc> = updated_at.into();
+                tx.execute(
+                    "\
+					INSERT INTO offers (payment_id, pocket_id, fiat_currency, rate, exchanged_at, topup_value_minor_units, exchange_fee_minor_units, exchange_fee_rate_permyriad)\
+					VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8)",
+                    (payment_id, pocket_id, currency_code, rate, exchanged_at, topup_value_minor_units, exchange_fee_minor_units, exchange_fee_rate_permyriad),
+                )
+                .map_to_invalid_input("Failed to add new incoming payment to payments db")?;
```

Comment:
Error message doesn't seem to fit
```suggestion
                .map_to_invalid_input("Failed to add new incoming pocket offer to offers db")?;
```